### PR TITLE
812: Leading and trailing spaces in parameters lead to form error

### LINKED ISSUE #812: Leading and trailing spaces in parameters lead to form error
Works
> label=foo, value=bar
[good-form.xlsx](https://github.com/user-attachments/files/24601588/good-form.xlsx)

Does not work due to leading/training spaces around =
> label = foo, value = bar
[bad-form.xlsx](https://github.com/user-attachments/files/24601587/bad-form.xlsx)
> 
>  Error: [row : 2] Parameter 'value' has a value which is not valid. Values must begin with a letter or underscore. Subsequent characters can include letters, numbers, dashes, underscores, and periods.

Feels like we should trim or ignore the spaces

### PR DESCRIPTION
**Pull request by lindsay-stevens**
Closes #812

#### Why is this the best possible solution? Were any other approaches considered?

As outlined in the commit messages:

> change from regex etc to a lark grammar, which is easier to read
    and could support enhancements like quoted parameter values to
    handle values containing delimiters e.g. "a = 'b c', x = y"

Also goes some way towards #592 in the sense of organising how parameters are handled and tested. I think next steps there could look like 1) moving the parameter handling or validation code blocks out of xls2json.py into a different module(s) or classes, and 2) upgrading the error messages to the new pattern that includes the row number; but these seemed like deviating too far from the motivating issue #812.

#### What are the regression risks?

Primarily I think the changes to `constants.py`, but that module is not exported at the project level `__init__.py`. I otherwise made minimal compatibility changes to the existing tests to try and avoid regressions.

#### Does this change require updates to documentation? If so, please file an issue [here](https://github.com/XLSForm/xlsform.github.io) and include the link below.

Not directly, but I noticed that the current XLSForm reference template is missing many accepted parameters (perhaps intentionally). Hopefully collecting the parameter keys in one place will now make it easier to keep the docs in sync.

#### Before submitting this PR, please make sure you have:
- [x] included test cases for core behavior and edge cases in `tests`
- [x] run `python -m unittest` and verified all tests pass
- [x] run `ruff format pyxform tests` and `ruff check pyxform tests` to lint code
- [x] verified that any code or assets from external sources are properly credited in comments

### CODE REVIEW

File: tests/test_image_app_parameter.py [hunk 1]
```diff
@@ -48,31 +51,6 @@ def test_throwing_error_when_invalid_android_package_name_is_used_with_supported
                         ],
                     )
 
-    def test_throwing_error_when_blank_android_package_name_is_used_with_supported_appearances(
```
Comment (lognaturel) [inline tests/test_image_app_parameter.py:51]: Is this now just handled generically as a blank parameter? I have read this test a hundred times and I can't seem to convince myself it had any value so maybe that's why you removed it!

File: pyxform/xls2json.py [hunk 2]
```diff
@@ -1033,54 +1018,54 @@ def workbook_to_json(
...
-                    accepted=select_params_allowed,
-                    row_number=row_number,
-                )
+                    qt_params = constants.ParametersSelectFromFile
+                    pv.validate(
+                        parameters=parameters,
+                        accepted=qt_params,
+                        row_number=row_number,
+                    )
+                    if "value" in parameters:
```
Comment (lognaturel) [inline pyxform/xls2json.py]: Any particular reason not to introduce constants here and for label?

**Review by lognaturel — APPROVED** [review]
Nice! Couple of questions that could prompt action but overall this looks ready to merge and build on.

File: pyxform/xls2json.py (hunk 2 above)
Comment (lindsay-stevens) [reply pyxform/xls2json.py]: Just missed this bit (and also `text` param `rows) - fixed in 9fdd143

File: tests/test_image_app_parameter.py (hunk 1 above)
Comment (lindsay-stevens) [reply tests/test_image_app_parameter.py:51]: Yes this test was deleted because the case is a generic parsing failure, now handled by test_parameters.py (though that only had the first one case - added the second one in 9fdd143)

**Review by lognaturel — APPROVED** [review]
Fab!
